jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -67,10 +74,42 @@ class S3TreeTable(
     private val mouseListener = object : MouseAdapter() {
         override fun mouseClicked(e: MouseEvent) {
             val row = rowAtPoint(e.point).takeIf { it >= 0 } ?: return
+            handleOpeningFile(row, e)
             handleLoadingMore(row, e)
         }
     }
 
+    private fun handleOpeningFile(row: Int, e: MouseEvent) {
+        e.clickCount.takeUnless { it != 2 } ?: return
```

Comment:
nit: instead of the double-negative maybe `takeIf { it == 2 }` ? or even clearer might be 

```kt
if (e.clickCount != 2) return
```
(which is more consistent with `handleLoadingMore`

What happens if I accidentally triple-click?

Reply:
Changed. Good point, changed the logic to check for `clickCount >= 2` since if you click 3+ times we still want to do the action